dll/win32/shell32/wine/control.c

```diff
@@ -1173,20 +1173,70 @@ static	void	Control_DoLaunch(CPanel* panel, HWND hWnd, LPCWSTR wszCmd)
 void WINAPI Control_RunDLLW(HWND hWnd, HINSTANCE hInst, LPCWSTR cmd, DWORD nCmdShow)
 {
     CPanel	panel;
+#ifdef __REACTOS__
+    WCHAR *ptr1 = NULL;
+    WCHAR *ptr2 = NULL;
+    WCHAR comma = ',';
+    WCHAR dot = '.';
+    LPWSTR buffer;
+    WCHAR ext[4] = L"   ";
+    SIZE_T nLen = lstrlenW(cmd);
+    BOOL GoodExt;
+
+    TRACE("(%p, %p, %s, 0x%08x)\n",
+          hWnd, hInst, debugstr_w(cmd), nCmdShow);
+
+/* Search 'cmd' for first comma. If the remaining length is 1 or 2,
+ * then replace dot before extention with UNICODE_NULL. */
+
+    buffer = HeapAlloc(GetProcessHeap(), HEAP_ZERO_MEMORY, sizeof(*buffer) * (nLen + 1));
+    if (!buffer) return;
+
+    lstrcpyW(buffer, cmd);
+
+    ptr1 = wcschr(cmd, comma);
+    if (ptr1)
+        nLen = lstrlenW(ptr1+1);
+    else
+        nLen = 0;
+
+    ptr2 = wcschr(cmd, dot);
+    if (ptr2)
+    {
+        wcsncpy(ext, ptr2 + 1, 3);
+    }
+
+    GoodExt = wcsicmp(ext, L"DLL") == 0 || wcscmp(ext, L"CPL") == 0;
```

Comment:
The second one should probably also be `wcsicmp` ?

Reply:
I agree. Changed. Thanks.

Reply:
Now I have completely removed the second condition. Thanks.